src/@types/casperlabsSigner.d.ts

```diff
@@ -17,16 +17,21 @@ interface CasperLabsHelper {
   requestConnection: () => void;
 
   /**
-   * Send Deploy in JSON format message to Signer plugin to sign.
+   * Send Deploy in JSON format to Signer extension to be signed.
    *
-   * @param deploy deploy in JSON format
-   * @param sourcePublicKeyHex public key in hex format with algorithm prefix. Used to sign the deploy
-   * @param targetPublicKeyHex public key in hex format with algorithm prefix. Used to display hex-formatted address on the UI
+   * @param deploy - deploy in JSON format
+   * @param signingPublicKeyHex - Hex-formatted public key. The corresponding secret key is used to sign the deploy.
+   * @param {string} [targetPublicKeyHex] - Hex-formatted public key.
+   * If the `target` in the deploy is an account hash this can be used to verify it and display the hex-formatted public key in the UI.
+   *
+   * @throws Errors if the Signer extension is not connected.
+   * @throws Errors if signingPublicKey is not available or does not match the Active Key in the Signer.
+   * @throws Errors if targetPublicKeyHex is not the same as the key (or corresponding account hash) that is used as target in deploy.
    */
   sign: (
     deploy: { deploy: JsonTypes },
-    sourcePublicKeyHex: string,
-    targetPublicKeyHex: string
+    signingPublicKeyHex: string,
```

Comment:
what do you think about making `signingPublicKeyHex` parameter optional too? I believe we shouldn't need it anymore :)

Reply:
I don't mind so I've raised it with the MAKE team to get their thoughts on it

Reply:
I think we'll leave it as is for now and we can make that change in future, the logic around `signingPublicKeyHex` is slightly more involved.